app/helpers/template_helpers.py

```diff
@@ -94,6 +94,18 @@ def service_links_context(
 
         return None
 
+    @property
+    def data_layer_context(
+        self,
+    ) -> Union[list[dict], None, list]:
+        if metadata := get_metadata(current_user):
+            if tx_id := metadata.get("tx_id"):
+                return self._survey_config.get_data_layer(
+                    tx_id=tx_id,
+                )
+
+        return self._survey_config.get_data_layer()
```

Comment:
I think call the function once and pass in a tx_id which may or may not be None. `tx_id` is also mandatory if we have metadata.

Reply:
Changed it to 
```python
tx_id = get_metadata(current_user).get("tx_id") if get_metadata(current_user) else None
```

Reply:
This would then call `get_metadata` twice, which should be avoided.

Reply:
Ok removed duplication now.